tests/test_image_transformer.py

```diff
@@ -1,93 +1,44 @@
-            doc.build(pdf_transformer.get_elements())
+        self.survey = json.loads(survey_data)
 
-            # Create page images from PDF
-            img_tfr = ImageTransformer(log, survey, reply)
-            images = list(img_tfr.create_image_sequence(fp, nmbr_seq=itertools.count()))
+        with open("./tests/data/EDC_134_20170301_1000.csv", "rb") as fb:
+            self.check = fb.read()
+        self.log = wrap_logger(logging.getLogger(__name__))
 
-            image_transformer = ImageTransformer(log, survey, reply)
+    def test_image_index(self):
 
-            # create the index file and capture creation date
-            with open(image_transformer.create_image_index(images), "r") as fh:
-                csv = fh.read()
-                date1 = csv.split(',')[0]
-
-            time.sleep(1)
-
-            # create the index file again and make sure the date is different
-            with open(image_transformer.create_image_index(images), "r") as fh:
-                csv = fh.read()
-                date2 = csv.split(',')[0]
-
-            self.assertNotEqual(date1, date2, "Image index dates should not be equal")
-
-    def test_mwss_index(self):
-
-        with open("./tests/data/eq-mwss.json") as fb:
-            data = fb.read()
-        reply = json.loads(data)
-
-        with open("./tests/data/134.0005.json") as fb:
-            survey_data = fb.read()
-        survey = json.loads(survey_data)
+        # Create page images from PDF
+        img_tfr = ImageTransformer(self.log, self.survey, self.reply, current_time=datetime.datetime(2017, 3, 7, 9, 45, 4))
 
-        with open("./tests/data/EDC_134_20170301_1000.csv") as fb:
-            check = fb.read()
+        img_tfr.get_zipped_images(itertools.count())
 
-        with tempfile.TemporaryDirectory(prefix="sdx_") as locn:
+        with img_tfr.index_file.in_memory_index as f:
+            csv = f.read()
 
-            # Build PDF
-            fp = os.path.join(locn, "pages.pdf")
-            doc = SimpleDocTemplate(fp, pagesize=A4)
-            pdf_transformer = PDFTransformer(survey, reply)
-            doc.build(pdf_transformer.get_elements())
+        self.assertEqual(self.check.splitlines(), csv.splitlines())
 
-            # Create page images from PDF
-            img_tfr = ImageTransformer(log, survey, reply)
-            images = list(img_tfr.create_image_sequence(fp, nmbr_seq=itertools.count()))
+    def test_current_time_not_set_at_initialisation(self):
 
-            with open(img_tfr.create_image_index(images, current_time=datetime.datetime(2017, 3, 7, 9, 45, 4)), "r") as fh:
-                csv = fh.read()
+        img_tfr1 = ImageTransformer(self.log, self.survey, self.reply)
+        time.sleep(0.01)
+        img_tfr2 = ImageTransformer(self.log, self.survey, self.reply)
 
-            self.assertEqual(check.splitlines(), csv.splitlines())
+        self.assertNotEqual(img_tfr1.current_time, img_tfr2.current_time)
```

Comment:
I think we could maybe have another test or a different test for this? I feel like assertNotEqual leaves it a bit open, although it's fine, just a though.

Reply:
The test is only checking that the time is not set to a constant at build time . So in that respect not equal I think suffices , since it makes sure that the time is different after the horrid little delay that was added

Reply:
👍